code/datums/status_effects/status_effect.dm

```diff
@@ -227,6 +230,14 @@
 	var/id = initial(path.identifier)
 	return LAZYACCESS(status_effects, id)
 
+/mob/proc/has_alerting_status_effect()
+	var/list_length = LAZYLEN(status_effects)
+	for(var/i = list_length;i>=0; i--)
+		var/datum/status_effect/S = LAZYACCESS(status_effects, i)
+		if(S.alert_admins_when_leaving)
+			return S
```

Comment:
This is a strange way to write this loop, it should probably be:
```suggestion
	for (var/datum/status_effect/S in status_effects)
		if (S.alert_admins_when_leaving)
			return S
```

Reply:
My fears what the for will do when the list is null, since that is the default value for the variable, not empty list

Reply:
`for (var/thing in null)` is valid, it's a no-op.